### DISCUSSION TITLE
Start with current version or wait for refactor?

### DISCUSSION THREAD

**Discussion by javea7171** [pocketbase/pocketbase#5543] [Q&A] [answered]
Hello, I'm wondering whether I should start now with pocketbase or wait for the refactoring to be completed before using it? Will there be an upgrade path from the current version to the refactored version?

**Comment by ganigeorgiev** [comment, marked as answer]
It depends to what extend and how do you plan to use PocketBase. If you can pause your project or work on something else for around a month or two, then maybe you'll be better waiting for the refactoring, otherwise - prefer the latest v0.22.x release.

The upcoming Web APIs breaking changes are minor and for some frequently used user endpoints (eg. `listAuthMethods()`) we'll even return both the old and new responses (because not always you can enforce client-side app update at the same time with the server, e.g. on a mobile app).

There will be major breaking changes to the Go and JSVM (pb_hooks/pb_migrations) APIs.
An upgrade guide listing all changes will be available but it will require users to perform manual code migration steps (only the `pb_data` will be automigrated).

I'm at the end of the refactoring and currently testing and polishing the changes by migrating locally https://github.com/presentator/presentator (it took me ~1.5-2h to adjust the custom Go code and its tests).

My todo has couple more things to complete before I feel confident publishing a release candidate but there are no fixed ETAs when that will be and I don't want to rush it as it is very easy to miss a minor bug that can cause a major security issue later.
